enclave/core/init_fini.c

```diff
@@ -111,14 +133,35 @@ void oe_call_init_functions(void)
 **==============================================================================
 */
 
-void oe_call_fini_functions(void)
+static void _call_fini_functions(
+    void (**fini_array_start)(void),
+    void (**fini_array_end)(void))
 {
     void (**fn)(void);
+
+    for (fn = fini_array_end - 1; fn >= fini_array_start; fn--)
+    {
+        (*fn)();
+    }
+}
+
+void oe_call_fini_functions(void)
+{
     extern void (*__fini_array_start)(void);
     extern void (*__fini_array_end)(void);
+    const uint64_t baseaddr = (uint64_t)__oe_get_enclave_base();
+    const oe_enclave_module_info_t* module_info = oe_get_module_info();
 
-    for (fn = &__fini_array_end - 1; fn >= &__fini_array_start; fn--)
+    _call_fini_functions(&__fini_array_start, &__fini_array_end);
```

Comment:
This ordering is fine for FIPS. For general multi-module loading, we'd need to think about this carefully. Lot of the core functionality is provided in `oecore`, and we'd want to make sure that that is the root-most module so that it's destructors, if any, get called last.
Maybe that means that our current static libraries ought to be converted to shared libraries. Needs more thought.